Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp

```diff
@@ -1014,20 +1041,22 @@ RefPtr<WebCore::ShapeDetection::TextDetector> WebChromeClient::createTextDetecto
 
 void WebChromeClient::attachRootGraphicsLayer(LocalFrame& frame, GraphicsLayer* layer)
 {
+    auto page = protectedPage();
     if (layer)
-        m_page.enterAcceleratedCompositingMode(frame, layer);
+        page->enterAcceleratedCompositingMode(frame, layer);
     else
-        m_page.exitAcceleratedCompositingMode(frame);
+        page->exitAcceleratedCompositingMode(frame);
 }
 
 void WebChromeClient::attachViewOverlayGraphicsLayer(GraphicsLayer* graphicsLayer)
 {
-    auto* drawingArea = m_page.drawingArea();
+    auto page = protectedPage();
+    auto* drawingArea = page->drawingArea();
```

Comment:
Surprised this doesn't need to be a smart pointer.

Reply:
DrawingArea is not currently ref-counted.